internal/certs/certs.go

```diff
@@ -146,39 +77,3 @@ func PEMEncodeCertificate(raw []byte) []byte {
 func pemEncodePrivateKey(raw []byte) []byte {
 	return pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: raw})
 }
-
-func newCA() (*x509.Certificate, *rsa.PrivateKey, error) {
```

Comment:
No longer used, we'll rely on helm or the user to provide the CA.

Reply:
Right, good question. I added another TODO to the list in the description. We need to generate the CA in helm as part of the deploy. That should make it invisible to the user. They won't have to do anything new, but helm will take care of creating the secret which contains the CA and CA private key.

Reply:
I've pushed a commit which does this (generates the CA in helm and stores it in a secret)

Reply:
The other implication is that the server can no longer be served outside of a helm install, e.g. docker, compose, or local build, unless the CA is manually created and configured. It'll be nice to keep this to support those use cases but not critical

Reply:
Ya, that's true. We could keep the ability to generate a CA if none is provided for development and demo purposes. I think in practice no real production use case should have the server generate a CA on first install. The operator really needs to save that CA and private key somewhere, and they aren't going to know it exists or where to find it if it was generated on first start.

I'll see about re-adding this for development.

Reply:
One of the disadvantages to generating the CA by default is that if the user accidentally mis-configured either `ca` or `caPrivateKey`. For example, maybe they set only one of the two, or they set `primaryKey` instead of `caPrimaryKey`. With the changes in this PR they will get a nice error and know how to fix it.

If we always generate things for them by default then it won't fail, and they'll have a harder time debugging.  

I think it might be better to require these are set ahead of starting the server. For dev maybe we could add a `--dev` flag and do some setup in the CLI before starting the server.

Reply:
Either Helm or infra can exit error if only one of `ca` or `caPrivateKey` is supplied since both are required. It'll reduce the impact of the scenario you're suggesting